ChiModules/LinearBoltzmannSolver/Acceleration/diffusion_mip_02a_assembleAand_b.cc

```diff
@@ -0,0 +1,377 @@
+
+      std::vector<double> qg(num_nodes, 0.0);
+      for (size_t j=0; j<num_nodes; j++)
+        qg[j] = q_vector[m_sdm.MapDOFLocal(cell, j, m_uk_man, 0, g)];
+
+      //==================================== Assemble continuous terms
+      for (size_t i=0; i<num_nodes; i++)
+      {
+        const int64_t imap = m_sdm.MapDOF(cell,i,m_uk_man,0,g);
+        double entry_rhs_i = 0.0;
+        for (size_t j=0; j<num_nodes; j++)
+        {
+          const int64_t jmap = m_sdm.MapDOF(cell,j,m_uk_man,0,g);
+          double entry_aij = 0.0;
+          for (size_t qp : qp_data.QuadraturePointIndices())
+          {
+            entry_aij += Dg *
+              qp_data.ShapeGrad(i, qp).Dot(qp_data.ShapeGrad(j, qp)) *
+              qp_data.JxW(qp);
+
+            entry_aij += sigr_g *
+              qp_data.ShapeValue(i, qp) * qp_data.ShapeValue(j, qp) *
+              qp_data.JxW(qp);
+
+            if (source_function.empty())
+              entry_rhs_i +=
+                qg[j] *
+                qp_data.ShapeValue(i, qp) * qp_data.ShapeValue(j, qp) *
+                qp_data.JxW(qp);
+            else
+            {
+              entry_rhs_i +=
+                CallLuaXYZFunction(L, source_function, qp_data.QPointXYZ(qp)) *
+                qp_data.ShapeValue(i, qp) * qp_data.ShapeValue(j, qp) *
+                qp_data.JxW(qp);
+            }
+          }//for qp
+          MatSetValue(m_A, imap, jmap, entry_aij, ADD_VALUES);
+        }//for j
+
+        VecSetValue(m_rhs, imap, entry_rhs_i, ADD_VALUES);
+      }//for i
+
+      //==================================== Assemble face terms
+      for (size_t f=0; f<num_faces; ++f)
+      {
+        const auto&  face           = cell.faces[f];
+        const auto&  n_f            = face.normal;
+        const size_t num_face_nodes = cell_mapping.NumFaceNodes(f);
+        const auto   fqp_data   = cell_mapping.MakeFaceQuadraturePointData(f);
+
+        const double hm = HPerpendicular(cell, f);
+
+        typedef chi_mesh::MeshContinuum Grid;
+
+        if (face.has_neighbor)
+        {
+          const auto&  adj_cell         = m_grid.cells[face.neighbor_id];
+          const auto&  adj_cell_mapping = m_sdm.GetCellMapping(adj_cell);
+          const auto   ac_nodes         = adj_cell_mapping.GetNodeLocations();
+          const size_t acf              = Grid::MapCellFace(cell, adj_cell, f);
+          const double hp               = HPerpendicular(adj_cell, acf);
+
+          const auto&  adj_xs   = m_map_mat_id_2_xs.at(adj_cell.material_id);
+          const double adj_Dg   = adj_xs.Dg[g];
+
+          //========================= Compute kappa
+          double kappa = 1.0;
+          if (cell.Type() == chi_mesh::CellType::SLAB)
+            kappa = fmax(2.0*(adj_Dg/hp + Dg/hm),0.25);
+          if (cell.Type() == chi_mesh::CellType::POLYGON)
+            kappa = fmax(2.0*(adj_Dg/hp + Dg/hm),0.25);
+          if (cell.Type() == chi_mesh::CellType::POLYHEDRON)
```

Comment:
maybe add a comment that the penalty constant is larger here.  In the end, it could be a constant supplied by the user. What do you think?